pkg/buildkit/buildkit.go

```diff
@@ -256,31 +253,31 @@ func (c *Client) solveWith(modify func(buildDir string, solveOpt *bkclient.Solve
 		return err
 	}
 
-	return c.runSolve(solveOpt)
+	return c.runSolve(ctx, solveOpt)
 }
 
-func (c *Client) runSolve(so bkclient.SolveOpt) error {
+func (c *Client) runSolve(ctx context.Context, so bkclient.SolveOpt) error {
 	lw := &LogWriter{Logger: c.log}
 	ch := make(chan *bkclient.SolveStatus)
-	eg, ctx := errgroup.WithContext(c.ctx)
+	eg, ctx := errgroup.WithContext(ctx)
 
 	eg.Go(func() error {
-		if _, err := c.bk.Solve(ctx, nil, so, ch); err != nil {
-			return err
+		var c console.Console
+		if cn, err := console.ConsoleFromFile(os.Stderr); err != nil {
+			c = cn
 		}
 
-		c.log.Info("Solve complete")
-		return nil
+		_, err := progressui.DisplaySolveStatus(context.TODO(), "", c, lw, ch)
```

Comment:
Why not `ctx` here like before?

Reply:
Great question. That was my original design but I noticed that sometimes the build would hang indefinitely after cancellation. A little digging revealed that, if the progress printer is cancelled before solve returns, then a routine inside the latter will block when writing to the status channel since there is nothing left to read from it and the `Wait()` func would hang indefinitely.

Details here: https://github.com/moby/buildkit/pull/1721

Reply:
i added a quick change and note to indicate the reason why this was done. what do you think?